### PR TITLE
fix(exchange.account): broken CSV file

### PR DESCRIPTION
**Pull request by y4nnL**
| Question         | Answer
| ---------------- | ---
| Branch?          | `MANAGER-8230`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix #MANAGER-8716
| License          | BSD 3-Clause

- [X] Try to keep pull requests small so they can be easily reviewed.
- [X] Commits are signed-off
- [ ] ~~Only FR translations have been updated~~
- [X] Branch is up-to-date with target branch
- [X] Lint has passed locally
- [X] Standalone app was ran and tested locally
- [X] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description

When exporting exchange accounts to CSV, the file is completely broken due to malformed fields

### CODE REVIEW

File: packages/manager/modules/exchange/src/account/export-as-csv/account-export-as-csv.controller.js [hunk 1]
```diff
@@ -90,6 +90,8 @@ export default class ExchangeExportToCsvAccountsCtrl {
         'samaccountName',
         'taskPendingId',
         'id',
+        // https://answers.microsoft.com/fr-fr/msoffice/forum/all/excel-sortie-en-csv-avec-conversion-utf-8/0a05dc63-645d-4303-b0f5-c27513bf776e
```
Comment (marie-j) [inline packages/manager/modules/exchange/src/account/export-as-csv/account-export-as-csv.controller.js:93]: Link to an issue/solution written in english would be better 
Also why is this added above a property that doesn't seems to be transformed ?

File: packages/manager/modules/exchange/src/account/account.constants.js [hunk 2]
```diff
@@ -6,19 +6,51 @@ export const ACCOUNT_COUNTRY_RULE_NAME = 'country';
...
+    String: (v) => v,
+    Number: (v) => `${v}`,
+    Boolean: (v) => `${v}`,
+    Undefined: () => '',
+    Null: () => '',
+    Object: (v) => toStringMap.Array(Object.values(v)),
+    Array: (v) => v.map(accountValuetoCSVString).join(','),
+    Date: (v) => v.toISOString(),
+  };
+  const type = Object.prototype.toString.call(value).match(/ (.+?)\]$/)[1];
```
Comment (marie-j) [inline packages/manager/modules/exchange/src/account/account.constants.js:30]: `typeof` is not suiting the needs here ?

**Review by marie-j — CHANGES_REQUESTED** [review]

File: packages/manager/modules/exchange/src/account/account.constants.js [hunk 3]
```diff
@@ -6,19 +6,51 @@ export const ACCOUNT_COUNTRY_RULE_NAME = 'country';
...
+ * Legacy
+ * @see MANAGER-8716
+ */
 export const unescapeDescription = (description) =>
   typeof description === 'string' ? description.replace(/^"+|"+$/g, '') : null;
 
+const accountPropertiesWithUnit = ['totalQuota', 'usedQuota'];
+const accountPropertiesWithStatus = ['mfa'];
+
+const accountValuetoCSVString = (value) => {
```
Comment (radireddy) [inline packages/manager/modules/exchange/src/account/account.constants.js:19]:
```suggestion
const accountValueToCSVString = (value) => {
```

File: packages/manager/modules/exchange/src/account/export-as-csv/account-export-as-csv.controller.js (hunk 1 above)
Comment (y4nnL) [reply packages/manager/modules/exchange/src/account/export-as-csv/account-export-as-csv.controller.js:93]: None of these properties are transformed, there are simply removed from the output for different reasons
The description field breaks the output file that's why it is removed. I'll find an english version of this issue :+1:

File: packages/manager/modules/exchange/src/account/account.constants.js (hunk 2 above)
Comment (y4nnL) [reply packages/manager/modules/exchange/src/account/account.constants.js:30]: Nope, typeof is not precise enough for some of the types (Array, Date and null will ouput "object")
Object.prototype.toString is more reliable as it gives the native object / primitive in the form of [object Xxxx]

**Review by radireddy — APPROVED** [review]

**Review by marie-j — APPROVED** [review]
